lib/galaxy/datatypes/microarrays.py

```diff
@@ -125,3 +126,66 @@ def set_meta(self, dataset, **kwd):
         dataset.metadata.number_of_optional_header_records = int(headers[1][0])
         dataset.metadata.number_of_data_columns = int(headers[1][1])
         dataset.metadata.file_type = headers[2][0].strip().strip('"').split("=")[1]
+
+
+class Cel(Binary):
+    """ Cel File format described at:
+            http://media.affymetrix.com/support/developer/powertools/changelog/gcos-agcc/cel.html
+    """
+
+    file_ext = "cel"
+    edam_format = "format_1638"
+    edam_data = "data_3110"
+    MetadataElement(name="version", default="3", desc="Version", readonly=True, visible=True,
+                    optional=True, no_value="3")
+
+    def sniff(self, filename):
+        """
+        Try to guess if the file is a Cel file.
+        >>> from galaxy.datatypes.sniff import get_test_fname
+        >>> fname = get_test_fname('affy_v_agcc.cel')
```

Comment:
Is `lib/galaxy/datatypes/test/affy_v_agcc.cel` actually needed? It seems of the same subtype of `test.cel` .

Can you instead add a test file for the version 4?

Reply:
Done